src/main/java/com/mongodb/hibernate/jdbc/MongoPreparedStatement.java

```diff
@@ -66,15 +71,16 @@ public ResultSet executeQuery() throws SQLException {
         checkClosed();
         closeLastOpenResultSet();
         checkAllParametersSet();
-        return executeQueryCommand(command);
+        return executeQuery(command);
```

Comment:
`checkSupportedQueryCommand` is missing here, and it should be after `checkAllParametersSet`, similarly to how it is in `executeUpdate`.

```suggestion
        checkSupportedQueryCommand(command);
        return executeQuery(command);
```